docs/source/algorithms/MaskBinsIf-v1.rst

```diff
@@ -0,0 +1,35 @@
+
+.. algorithm::
+
+.. summary::
+
+.. relatedalgorithms::
+
+.. properties::
+
+Description
+-----------
+
+This algorithm masks bins according to the criteria specified as a `muparser <http://beltoforion.de/article.php?a=muparser>`_ expression.
+The variables entering the criteria are reserved as follows:
+
+- y : count in a bin
+- x : the bin center
+- e : the error on the count
+- dx : the error on the bin center
+- i : the workspace index
```

Comment:
Runs from 0 - number of spectrum -1. Not to be confused with the spectrum number.

As a related point, we hoped to make an effort to increase (mandate) the useage of indexes internal to Mantid, but keep (prefer) ids externally. Would it be better to change I to be the Spectrum Number. Gives us better protection from operations which affect i, such as cropping the workspace, or extracting spectra?

Reply:
The spectrum number can be specified with 's' as a spectrum axis value, isn't it? If the spectrum axis is converted though, then I don't think the spectrum number makes much sense. So I thought the combination of workspace index and the spectrum axis value covers all.

Reply:
This is a little tricky. Worth discussing I think as once published, very hard to roll-back.

In the case of the x-axis you use the actual values, not the bin indices. For spectrum axis converted to say two theta, can't we do the same?

Reply:
It is the same for the spectrum axis; 's' is the value of the axis vector. But if the axis is not converted to anything, the axis value is the spectrum number, isn't it?

Reply:
@gvardany I think so. If this is the case, can we keep `s` but drop the `i`? Does it serve any useful purpose that is needed now?

Reply:
Yes, we can drop `i`; it was more like for convenience.

Reply:
Actually, reading the code I realised that there might be a use case for `i`; when the spectrum axis is a text axis.

Reply:
If you have a `text` axis, would you still ever want to apply a masking operation?

Reply:
I really don't know the "right" answers, I'm just trying to think it through as well.

Reply:
I'm not sure, one could have text labels for the spectra, but still wish to mask some of them. So in principle this can be used although I haven't seen text axes in use for a long time.

Reply:
I'm slightly worried that the following (type) mask operations will do different things when using the `i` notation:

```
 ws = CreateSampleWorkspace(BankPixelWidth=100, NumBanks=1, OutputWorkspace='out')
 ws =  MaskBinsIf(InputWorkspace=ws, Criterion='i >10 && i<20 && x>1000 && x<2000', OutputWorkspace='out')
ws = CropWorkspace(ws, StartWorkspaceIndex=0, EndWorkspaceIndex=10)
 ws =  MaskBinsIf(InputWorkspace=ws, Criterion='i >10 && i<20 && x>1000 && x<2000', OutputWorkspace='out')
```

One option would be to refuse to process any workspace that does not have a numeric vertical axis. That way you can only provide `s`. We could add `i` at a later point if really required?

Reply:
That sounds sensible I guess.

Reply:
If you're really dead set on passing indexes in, I would be ok with that. I'm just wary of it having made the mistake myself a few times.